tests/e2e/bcd_consumer_integration_test.go

```diff
@@ -402,31 +402,47 @@ func (s *BCDConsumerIntegrationTestSuite) Test7ConsumerFPRewards() {
 	s.Empty(balance)
 
 	// Commit public randomness at the activated block height on the consumer chain
-	randListInfo, msgCommitPubRandList, err := datagen.GenRandomMsgCommitPubRandList(r, czFpBTCSK, uint64(czActivatedHeight), 100)
+	numPubRand := uint64(200)
+	randListInfo, msgCommitPubRandList, err := datagen.GenRandomMsgCommitPubRandList(r, czFpBTCSK, uint64(czActivatedHeight), numPubRand)
 	s.NoError(err)
 
 	// Submit the public randomness to the consumer chain
 	txResp, err := s.cosmwasmController.CommitPubRandList(
 		czFpBTCPK,
 		uint64(czActivatedHeight),
-		100,
+		numPubRand,
 		randListInfo.Commitment,
 		msgCommitPubRandList.Sig.MustToBTCSig(),
 	)
 	s.NoError(err)
 	s.NotNil(txResp)
 
-	// Consumer finality provider submits finality signature
-	txResp, err = s.cosmwasmController.SubmitFinalitySig(
-		czFpBTCSK,
-		czFpBTCPK,
-		randListInfo.SRList[0],
-		&randListInfo.PRList[0],
-		randListInfo.ProofList[0].ToProto(),
-		czActivatedHeight,
-	)
-	s.NoError(err)
-	s.NotNil(txResp)
+	// Consumer finality provider submits finality signature, eventually
+	//succeeds
+	s.Eventually(func() bool {
+		// TODO: Obtain or compute the exact epoch to be finalised to cover the
+		// public randomness commit
+		finalizeEpoch := uint64(20)
+		s.finalizeUntilEpoch(finalizeEpoch, false)
```

Comment:
I'm not sure if this function is suitable for being triggered repetitively in an `Eventual` function.

What do you think we 

- after committing pub rand, check the height of `SRList[0]` is not finalised, and then submit a finality sig and expect error
- call `finalizeUntilEpoch` and then try submitting finality sig again, and expect no error